packages/react/src/redux.ts

```diff
@@ -89,6 +96,15 @@ function createReduxEnhancer(enhancerOptions?: Partial<SentryEnhancerOptions>):
 
   return (next: StoreEnhancerStoreCreator): StoreEnhancerStoreCreator =>
     <S = any, A extends Action = AnyAction>(reducer: Reducer<S, A>, initialState?: PreloadedState<S>) => {
+      options.attachReduxState &&
+        addGlobalEventProcessor((event, hint) => {
+          hint.attachments = [
+            ...(hint.attachments || []),
+            { filename: 'reduxState.json', data: JSON.stringify(event.contexts && event.contexts.state) || ' ' },
+          ];
+          return event;
+        });
+
```

Comment:
Let's check before even if we have a state not to add an empty JSON attachment
```suggestion
      options.attachReduxState &&
      event.contexts && 
      event.contexts.state &&
        addGlobalEventProcessor((event, hint) => {
          hint.attachments = [
            ...(hint.attachments || []),
            { filename: 'reduxState.json', data: JSON.stringify(event.contexts.state) },
          ];
          return event;
        });

```

Reply:
To prevent issues with an undefined 'event' object on lines 110 and 111, I've suggested the below code, which has a simple check. The Redux state attachment is only added if 'event.contexts' and 'event.contexts.state' are defined, ensuring consistent behavior. What do you think? @HazAT 

```typescript
options.attachReduxState &&
addGlobalEventProcessor((event, hint) => {
  if (event.contexts && event.contexts.state) {
    hint.attachments = [
      ...(hint.attachments || []),
      { filename: 'reduxState.json', data: JSON.stringify(event.contexts.state) },
    ];
  }
  return event;
});

```

Reply:
We can even check if `event.contexts.state` is of type `redux` to be safer - but your direction is correct @malay44